src/main/java/pkmncore/PokemonManager.java

```diff
@@ -0,0 +1,67 @@
+package pkmncore;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PokemonManager {
+
+    private StorageUnit storage;
+
+    public PokemonManager(StorageUnit storage) {
+        this.storage = storage;
+    }
+
+    public void catchPokemon(Pokemon pokemon) throws PokemonError {
+        checkForNullPokemon(pokemon);
+        savePokemon(pokemon);
+    }
+
+    public List<Pokemon> viewCaughtPokemon() {
+        List<Pokemon> caughtPokemon = new ArrayList<Pokemon>();
+        List<List<String>> allPokemon = getPokemon();
+        for (List<String> pokemonDetails : allPokemon) {
+            getDetails(caughtPokemon, pokemonDetails);
+        }
+        return caughtPokemon;
+    }
+
+    private void getDetails(List<Pokemon> caughtPokemon, List<String> pokemonDetails) {
+        String name = pokemonDetails.remove(0);
+        String height = pokemonDetails.remove(0);
+        String[] abilities = getAbilities(pokemonDetails);
+        Pokemon pokemon = new NamedPokemon(name, height, abilities);
+        caughtPokemon.add(pokemon);
```

Comment:
Same issue as the other methods that return nothing but mutate a list.
Just have them return a fresh list of things and either re-assign it outside or just do an 
firstList.addAll(convertDetails(pokeDetails));